### PR TITLE
Update address validation - Closes #832

### LINKED ISSUE #832: Update validation for address to match lisk-core
### Expected behavior
- Address which includes leading zero has to be invalid.
- Address include `.` should be invalid

https://github.com/LiskHQ/lisk/commit/a1fe9a9f2e3f85cb7a6e3ad3ba6ab9e51863327c

### Actual behavior
It allows to create transaction using above pattern.

https://github.com/LiskHQ/lisk-elements/blob/master/src/transactions/utils/validation.js#L51

### Which version(s) does this affect? (Environment, OS, etc...)
1.0-

### PR DESCRIPTION
**Pull request by mitsujutsu**
### What was the problem?

Address validation allowed leading zeros and inclusion of `.` character.

### How did I fix it?

Updated `validateAddress` in utils.

### How to test it?

npm run test

### Review checklist

* The PR resolves #832 
* All new code is covered with unit tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated

### CODE REVIEW

File: packages/lisk-transactions/test/utils/validation/validation.ts [hunk 1]
```diff
@@ -247,6 +257,15 @@ describe('validation', () => {
 				return expect(validateAddress.bind(null, address)).to.throw(error);
 			});
 		});
+
+		describe('Given an address that has leading zeros', () => {
+			const address = '00015133512790761431L';
+			const error = 'Address number does not have natural representation.';
+
+			it('should throw', () => {
```
Comment (ishantiw) [inline packages/lisk-transactions/test/utils/validation/validation.ts]: maybe `should throw an error` sounds better

File: packages/lisk-transactions/test/utils/validation/validation.ts [hunk 2]
```diff
@@ -238,6 +238,16 @@ describe('validation', () => {
 			});
 		});
 
+		describe('Given an address that includes `.`', () => {
+			const address = '14.15133512790761431L';
+			const error =
+				'Address format does not match requirements. Address includes invalid character: `.`.';
+
+			it('should throw', () => {
```
Comment (ishantiw) [inline packages/lisk-transactions/test/utils/validation/validation.ts]: maybe `should throw an error` sounds better

File: packages/lisk-transactions/src/utils/validation/validation.ts [hunk 3]
```diff
@@ -67,6 +67,7 @@ export const validateKeysgroup = (keysgroup: ReadonlyArray<string>) => {
 
 const MIN_ADDRESS_LENGTH = 2;
 const MAX_ADDRESS_LENGTH = 22;
+const BASE_TEN = 10;
 export const validateAddress = (address: string) => {
```
Comment (shuse2) [inline packages/lisk-transactions/src/utils/validation/validation.ts]: I just realized, but this should have return type

File: packages/lisk-transactions/test/utils/validation/validation.ts [hunk 4]
```diff
@@ -247,6 +257,15 @@ describe('validation', () => {
 				return expect(validateAddress.bind(null, address)).to.throw(error);
 			});
 		});
+
+		describe('Given an address that has leading zeros', () => {
+			const address = '00015133512790761431L';
+			const error = 'Address number does not have natural representation.';
```
Comment (shuse2) [inline packages/lisk-transactions/test/utils/validation/validation.ts]: `error` => `errorMessage`?

File: packages/lisk-transactions/src/utils/validation/validation.ts [hunk 5]
```diff
@@ -83,14 +84,25 @@ export const validateAddress = (address: string) => {
...
+	const addressNumber = new BigNum(addressString);
+
+	if (addressNumber.cmp(new BigNum(MAX_ADDRESS_NUMBER)) > 0) {
 		throw new Error(
 			'Address format does not match requirements. Address out of maximum range.',
 		);
 	}
 
+	if (addressString !== addressNumber.toString(BASE_TEN)) {
+		throw new Error('Address number does not have natural representation.');
```
Comment (shuse2) [inline packages/lisk-transactions/src/utils/validation/validation.ts]: I'm not sure about the "natural representation".
Maybe better to say directly that string format doesn't match the number?

Comment (mitsujutsu) [reply packages/lisk-transactions/src/utils/validation/validation.ts]: This is the wording used in core, but I agree not very clear.

**Pull request merged**